app/util/HelperFunctions.scala

```diff
@@ -14,4 +16,34 @@ object HelperFunctions {
     val t = AtomType.valueOf(atomType)
     Either.cond(t.isDefined, t.get, InvalidAtomTypeError)
   }
+}
+
+object Updater {
+  def update(map: Map[String, Any], path: String, value: JsLookupResult): Either[AtomAPIError, Map[String, Any]] =
+  try {
+    Right(updateNestedMap(map, path.split('.').toList, value.as[String]))
+  } catch {
+    case e: AtomAPIError => Left(e)
+  }
+
+  private def updateNestedMap[T](map: Map[String, Any], path: List[String], value: T): Map[String, Any] = path match {
+    case key :: Nil =>
+      // These checks shouldn't be necessary when using proper validation
+      if(map(key).isInstanceOf[Option[Any]]) {
+        map.updated(key, Some(value))
+      } else if(value.getClass != map(key).getClass) {
+        Logger.error("Types not matching when trying to update.")
+        throw new WrongTypeFound(found = value.getClass.getCanonicalName, expected = map(key).getClass.getCanonicalName)
```

Comment:
So far I've tried to avoid throwing errors in the API, instead using eithers and returning them as a json object to the client. I can tidy this up whilst doing the merge!